### ISSUE TITLE
Can't create application UIWindow without resorting to runtime arguments because CGRect doesn't like being a component

### ISSUE THREAD

**Issue by fatuhoku** [appsquickly/Typhoon#456]
The [standard](https://medium.com/@austinlouden/your-first-ios-app-100-programmatically-67cc41319bd8#.x8jhke28g) [way](https://developer.apple.com/library/ios/documentation/WindowsViews/Conceptual/ViewPG_iPhoneOS/CreatingWindows/CreatingWindows.html) to initialize a `UIWindow` for your application looks like this:

``` objc
self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
```

I wanted Typhoon to be right on the top level of everything — and wanted the window to be injected directly into the AppDelegate. I tried doing this:

``` objc
#pragma mark - Screen & Window

- (UIScreen *)mainScreen {
    return [TyphoonDefinition withFactory:[UIScreen class] selector:@selector(mainScreen)];
}

- (NSValue *)mainScreenBounds {  // doesn't work.
    return [TyphoonDefinition withFactory:[self mainScreen] selector:@selector(bounds)];
}

- (UIWindow *)window {
    return [TyphoonDefinition withClass:[UIWindow class] configuration:^(TyphoonDefinition *definition) {
        [definition useInitializer:@selector(initWithFrame:) parameters:^(TyphoonMethod *initializer) {
            [initializer injectParameterWith:[self mainScreenBounds]];
        }];
        [definition injectProperty:@selector(rootViewController) with:[self superRootViewController]];
    }];
}
```

... but all I got was a black screen. I know for a fact that it's the main screen bounds playing up after experimentation. What could be done about this?
#357 is definitely a related issue to this.

**Comment by igrekde** [comment]
Hi, @fatuhoku! Thanks for reporting, looks strange - I'll investigate it.

**Comment by spbvasilenko** [comment]
@etolstoy If issue not  investigated yet, I'll try it.

**Comment by etolstoy** [comment]
Solved by #518.
